dask_expr/_categorical.py

```diff
@@ -127,3 +134,39 @@ class AsUnknown(Elemwise):
     @functools.cached_property
     def _meta(self):
         return clear_known_categories(self.frame._meta)
+
+
+class Categorize(Blockwise):
+    _parameters = ["frame", "categories", "index"]
+    operation = staticmethod(_categorize_block)
+    _projection_passthrough = True
+
+    @functools.cached_property
+    def _meta(self):
+        meta = _categorize_block(
+            self.frame._meta, self.operand("categories"), self.operand("index")
+        )
+        return meta
+
+
+class GetCategories(ApplyConcatApply):
+    _parameters = ["frame", "columns", "index", "split_every"]
+
+    chunk = staticmethod(_get_categories)
+    aggregate = staticmethod(_get_categories_agg)
+
+    @property
+    def chunk_kwargs(self):
+        return {"columns": self.operand("columns"), "index": self.operand("index")}
+
+    @functools.cached_property
+    def _meta(self):
+        return ({}, pd.Series())
+
+    def _simplify_down(self):
+        return GetCategories(
+            Projection(self.frame, self.operand("columns")),
```

Comment:
This should protect against loops, e.g. if 

```
set(self.frame.columns) ==  set(self.operand("columns"))
```

then don't do anything

Reply:
Ah, right! 🤦